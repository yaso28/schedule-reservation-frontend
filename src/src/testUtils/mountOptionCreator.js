import { createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import routes from '@/router/routes'
import Vuex from 'vuex'
import VueI18n from 'vue-i18n'
import i18nOptions from '@/i18n/options'
import VueMeta from 'vue-meta'
import CoreuiVue from '@coreui/vue'
import icons from '@/icons'
import services from '@/services'

class MountOptionCreator {
  create (storeModules = null) {
    const localVue = createLocalVue()

    // router
    localVue.use(VueRouter)
    const router = new VueRouter({
      mode: 'history',
      base: process.env.BASE_URL,
      routes
    })

    // i18n
    localVue.use(VueI18n)
    const i18n = new VueI18n(i18nOptions)

    // vue-meta
    localVue.use(VueMeta)

    // Core UI
    localVue.use(CoreuiVue)

    const option = {
      localVue,
      router,
      i18n,
      icons,
      provide: services
    }

    // store
    if (storeModules) {
      localVue.use(Vuex)
      option.store = new Vuex.Store({
        modules: storeModules
      })
    }

    return option
  }
}

export default new MountOptionCreator()
