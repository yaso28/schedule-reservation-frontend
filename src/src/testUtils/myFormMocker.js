import MyForm from '@/models/MyForm'
jest.mock('@/models/MyForm')

class MyFormMocker {
  createMockSingleField (field, initValue) {
    const input = {}
    input[field] = initValue
    return this.createMock(input)
  }

  createMock (input) {
    const mockForm = new MyForm(input)
    mockForm.values = input
    mockForm.errors = {}
    return mockForm
  }

  async setValue (wrapper, form, field, value) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        values: Object.assign({}, form.values, { [field]: value })
      })
    })
  }

  setValueNonProps (form, field, value) {
    form.values[field] = value
  }

  async setError (wrapper, form, field, errorMessage) {
    await wrapper.setProps({
      form: Object.assign({}, form, {
        errors: Object.assign({}, form.errors, { [field]: errorMessage })
      })
    })
  }
}

export default new MyFormMocker()
