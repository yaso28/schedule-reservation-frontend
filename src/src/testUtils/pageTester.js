import { mount } from '@vue/test-utils'
import MyPage from '@/components/containers/MyPage'

class PageTester {
  async mount (pageComponent, route, mountOption) {
    await mountOption.router.push(route)
    const pageWrapper = mount(MyPage, mountOption)
    return pageWrapper.findComponent(pageComponent)
  }

  assertTitleTag () {
    expect(document.title).toMatchSnapshot()
  }

  assertMetaTags () {
    expect(document.querySelectorAll('meta[data-vue-meta]')).toMatchSnapshot()
  }
}

export default new PageTester()
