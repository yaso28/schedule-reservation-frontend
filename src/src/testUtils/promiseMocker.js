import _ from 'lodash'

const createSetTimeoutPromise = getResult => new Promise((resolve, reject) => {
  setTimeout(() => getResult(resolve, reject), 1000)
})
const createSwitchPromise = (value, error, switchObj) => createSetTimeoutPromise(
  (resolve, reject) => switchObj.isSuccess ? resolve(value) : reject(error)
)
const createResolvedPromise = value => createSetTimeoutPromise(resolve => resolve(value))
const createRejectedPromise = error => createSetTimeoutPromise((resolve, reject) => reject(error))

const createMock = createPromise => jest.fn().mockImplementationOnce(createPromise)
const createMockForCases = (caseList, createPromiseForCurrentCase) => jest.fn((...args) => {
  for (let i = 0; i < caseList.length; i++) {
    const currentCase = caseList[i]
    if (_.isEqual(args, currentCase.args)) {
      return createPromiseForCurrentCase(currentCase)
    }
  }
  throw new Error('createMockForCases: no args matched')
})

class PromiseMocker {
  init () {
    jest.useFakeTimers()
  }

  clear () {
    jest.clearAllTimers()
  }

  flushPendings () {
    jest.runOnlyPendingTimers()
  }

  createSwitch () {
    return { isSuccess: true }
  }

  setSwitch (switchObj, isSuccess) {
    switchObj.isSuccess = isSuccess
  }

  mock (value, error, switchObj) {
    return createMock(() => createSwitchPromise(value, error, switchObj))
  }

  mockResolved (value) {
    return createMock(() => createResolvedPromise(value))
  }

  mockRejected (error) {
    return createMock(() => createRejectedPromise(error))
  }

  mockForCases (caseList) {
    return createMockForCases(caseList, currentCase => createSwitchPromise(currentCase.value, currentCase.error, currentCase.switch))
  }

  mockResolvedForCases (caseList) {
    return createMockForCases(caseList, currentCase => createResolvedPromise(currentCase.value))
  }

  mockRejectedForCases (caseList) {
    return createMockForCases(caseList, currentCase => createRejectedPromise(currentCase.error))
  }
}

export default new PromiseMocker()
