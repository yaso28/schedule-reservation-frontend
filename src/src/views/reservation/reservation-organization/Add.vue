<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="form"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MyText
            :label="$t('common.abbreviation')"
            :form="form"
            field="abbreviation"
          />
          <MyText
            :label="$t('reservation_organization.registration_number')"
            :form="form"
            field="registration_number"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ReservationOrganizationAdd',
  components: {
    MyText: require('@/components/inputs/MyText').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      form: null
    }
  },
  watch: {
    $route (to, from) {
      if (to.query.original_id + '' !== from.query.original_id + '') {
        this.load()
      }
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const originalId = this.$route.query.original_id
        const record = originalId ? await this.apiService.getReservationOrganization(originalId) : null
        this.form = this.apiService.makeForm({
          name: record?.name ?? '',
          abbreviation: record?.abbreviation ?? '',
          registration_number: record?.registration_number ?? ''
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.addReservationOrganization(this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.reservation-organization.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
