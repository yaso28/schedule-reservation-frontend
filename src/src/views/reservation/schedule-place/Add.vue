<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="form"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MyText
            :label="$t('common.abbreviation')"
            :form="form"
            field="abbreviation"
          />
          <MyNumber
            :label="$t('schedule_place.price_per_hour')"
            :form="form"
            field="price_per_hour"
            :min="0"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'SchedulePlaceAdd',
  components: {
    MyText: require('@/components/inputs/MyText').default,
    MyNumber: require('@/components/inputs/MyNumber').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      form: null
    }
  },
  watch: {
    $route (to, from) {
      if (to.query.original_id + '' !== from.query.original_id + '') {
        this.load()
      }
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const originalId = this.$route.query.original_id
        const record = originalId ? await this.apiService.getSchedulePlace(originalId) : null
        this.form = this.apiService.makeForm({
          name: record?.name ?? '',
          abbreviation: record?.abbreviation ?? '',
          price_per_hour: record?.price_per_hour ?? null
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.addSchedulePlace(this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.schedule-place.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
