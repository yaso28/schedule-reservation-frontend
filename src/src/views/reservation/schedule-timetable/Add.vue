<template>
  <div>
    <slot name="h1" />

    <CForm
      v-if="form"
      @submit.prevent="save"
    >
      <CCard>
        <CCardBody>
          <MyText
            :label="$t('common.name')"
            :form="form"
            field="name"
          />
          <MyTextarea
            :label="$t('schedule_timetable.details')"
            :form="form"
            field="details"
            :rows="10"
          />
        </CCardBody>
        <CCardFooter>
          <CButton
            id="btn-reset"
            color="secondary"
            @click="form.reset()"
          >
            {{ $t("action.reset") }}
          </CButton>
          <CButton
            type="submit"
            color="primary"
          >
            {{ $t('action.save') }}
          </CButton>
        </CCardFooter>
      </CCard>
    </CForm>
  </div>
</template>

<script>
export default {
  name: 'ScheduleTimetableAdd',
  components: {
    MyText: require('@/components/inputs/MyText').default,
    MyTextarea: require('@/components/inputs/MyTextarea').default
  },
  inheritAttrs: false,
  inject: ['apiService'],
  data () {
    return {
      form: null
    }
  },
  watch: {
    $route (to, from) {
      if (to.query.original_id + '' !== from.query.original_id + '') {
        this.load()
      }
    }
  },
  mounted () {
    this.load()
  },
  methods: {
    async load () {
      try {
        this.$store.commit('processing/beginGet')
        const originalId = this.$route.query.original_id
        const record = originalId ? await this.apiService.getScheduleTimetable(originalId) : null
        this.form = this.apiService.makeForm({
          name: record?.name ?? '',
          details: record?.details ?? ''
        })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endGet')
      }
    },
    async save () {
      try {
        this.$store.commit('processing/beginPost')
        this.$store.commit('alert/clear')
        await this.apiService.addScheduleTimetable(this.form)
        this.$store.dispatch('alert/saveSuccess')
        this.$router.push({ name: 'reservation.schedule-timetable.list' })
      } catch (error) {
        if (!this.apiService.isHandledError(error)) {
          throw error
        }
      } finally {
        this.$store.commit('processing/endPost')
      }
    }
  }
}
</script>
