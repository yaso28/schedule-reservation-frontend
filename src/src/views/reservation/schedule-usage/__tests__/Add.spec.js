import Add from '../Add'
import mountOptionCreator from '@/testUtils/mountOptionCreator'
import pageTester from '@/testUtils/pageTester'
import promiseMocker from '@/testUtils/promiseMocker'
import storeModules from '@/store/modules'
import myFormMocker from '@/testUtils/myFormMocker'

jest.mock('@/store/modules')
jest.mock('@/models/MyForm')

const reservationOrganizations = require('@/testUtils/dummyApiData/reservation-organization/list.json')
const originalRecord = require('@/testUtils/dummyApiData/schedule-usage/get.json')
const saveResult = require('@/testUtils/dummyApiData/common/add.json')

function setRouteCopyId (route, id) {
  route.query = { original_id: id }
}

describe('Add', () => {
  let mountOption
  let apiService
  let router
  let route
  beforeEach(() => {
    promiseMocker.init()
    mountOption = mountOptionCreator.create(storeModules)
    apiService = mountOption.provide.apiService
    router = mountOption.router
    route = { name: 'reservation.schedule-usage.add' }
  })

  afterEach(() => {
    promiseMocker.clear()
    jest.resetAllMocks()
  })

  describe('通常', () => {
    describe('マウント', () => {
      let wrapper
      let loadReservationOrganizationsSwitch
      let loadReservationOrganizationsError
      beforeEach(async () => {
        loadReservationOrganizationsSwitch = promiseMocker.createSwitch()
        loadReservationOrganizationsError = new Error('reservation organizations')
        apiService.getReservationOrganizationList = promiseMocker.mock(
          reservationOrganizations,
          loadReservationOrganizationsError,
          loadReservationOrganizationsSwitch
        )
        apiService.getScheduleUsage = jest.fn()
        wrapper = await pageTester.mount(Add, route, mountOption)
      })

      test('titleタグ', () => {
        pageTester.assertTitleTag()
      })

      test('metaタグ', () => {
        pageTester.assertMetaTags()
      })

      test('Processingカバー表示開始', () => {
        expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
      })

      test('予約団体マスタ取得処理を呼出', () => {
        expect(apiService.getReservationOrganizationList).toHaveBeenCalledWith()
      })

      test('コピー元データ取得処理は呼び出さない', () => {
        expect(apiService.getScheduleUsage).not.toHaveBeenCalled()
      })

      test('Processingカバー表示中', () => {
        expect(
          storeModules.processing.mutations.endGet
        ).not.toHaveBeenCalled()
      })

      describe('予約団体マスタ取得：エラー', () => {
        beforeEach(() => {
          promiseMocker.setSwitch(loadReservationOrganizationsSwitch, false)
          apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
          promiseMocker.flushPendings()
        })

        test('エラー処理', () => {
          expect(apiService.isHandledError).toHaveBeenCalledWith(loadReservationOrganizationsError)
        })

        test('Processingカバー表示終了', () => {
          expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
        })
      })

      describe('データ取得：成功', () => {
        let makeFormArgs
        let form
        beforeEach(() => {
          promiseMocker.setSwitch(loadReservationOrganizationsSwitch, true)
          makeFormArgs = {
            name: '',
            is_public: true,
            reservation_organization_id: null
          }
          form = myFormMocker.createMock(makeFormArgs)
          apiService.makeForm = jest.fn().mockReturnValueOnce(form)
          promiseMocker.flushPendings()
        })

        test('フォーム作成', () => {
          expect(apiService.makeForm).toHaveBeenCalledWith(makeFormArgs)
        })

        test('ページ全体表示', () => {
          expect(wrapper.element).toMatchSnapshot()
        })

        test('Processingカバー表示終了', () => {
          expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
        })

        describe('リセットボタンクリック', () => {
          beforeEach(() => {
            wrapper.find('#btn-reset').trigger('click')
          })

          test('フォームリセット', () => {
            expect(form.reset).toHaveBeenCalledWith()
          })
        })

        describe('フォーム送信', () => {
          let saveSwitch
          let saveError
          beforeEach(() => {
            saveSwitch = promiseMocker.createSwitch()
            saveError = new Error('save')
            apiService.addScheduleUsage = promiseMocker.mock(
              saveResult,
              saveError,
              saveSwitch
            )
            wrapper.find('form').trigger('submit')
          })

          test('Processingカバー表示開始', () => {
            expect(storeModules.processing.mutations.beginPost).toHaveBeenCalled()
          })

          test('アラートクリア', () => {
            expect(storeModules.alert.mutations.clear).toHaveBeenCalled()
          })

          test('保存処理を呼出', () => {
            expect(apiService.addScheduleUsage).toHaveBeenCalledWith(form)
          })

          test('Processingカバー表示中', () => {
            expect(
              storeModules.processing.mutations.endPost
            ).not.toHaveBeenCalled()
          })

          describe('保存：エラー', () => {
            beforeEach(() => {
              promiseMocker.setSwitch(saveSwitch, false)
              apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
              promiseMocker.flushPendings()
            })

            test('エラー処理', () => {
              expect(apiService.isHandledError).toHaveBeenCalledWith(saveError)
            })

            test('Processingカバー表示終了', () => {
              expect(storeModules.processing.mutations.endPost).toHaveBeenCalled()
            })
          })

          describe('保存：成功', () => {
            beforeEach(() => {
              promiseMocker.setSwitch(saveSwitch, true)
              router.push = jest.fn()
              promiseMocker.flushPendings()
            })

            test('アラート', () => {
              expect(storeModules.alert.actions.saveSuccess).toHaveBeenCalled()
            })

            test('リダイレクト', () => {
              expect(router.push).toHaveBeenCalledWith({ name: 'reservation.schedule-usage.list' })
            })

            test('Processingカバー表示終了', () => {
              expect(storeModules.processing.mutations.endPost).toHaveBeenCalled()
            })
          })
        })
      })
    })
  })

  describe('コピー', () => {
    let originalId
    beforeEach(() => {
      originalId = 1
      setRouteCopyId(route, originalId)
    })

    describe('マウント', () => {
      let loadSwitch
      let loadError
      let loadReservationOrganizationsSwitch
      let loadReservationOrganizationsError
      beforeEach(async () => {
        loadSwitch = promiseMocker.createSwitch()
        loadError = new Error('load')
        apiService.getScheduleUsage = promiseMocker.mock(
          originalRecord,
          loadError,
          loadSwitch
        )
        loadReservationOrganizationsSwitch = promiseMocker.createSwitch()
        loadReservationOrganizationsError = new Error('reservation organizations')
        apiService.getReservationOrganizationList = promiseMocker.mock(
          reservationOrganizations,
          loadReservationOrganizationsError,
          loadReservationOrganizationsSwitch
        )
        await pageTester.mount(Add, route, mountOption)
      })

      test('Processingカバー表示開始', () => {
        expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
      })

      test('予約団体マスタ取得処理を呼出', () => {
        expect(apiService.getReservationOrganizationList).toHaveBeenCalledWith()
      })

      test('コピー元データ取得処理を呼出', () => {
        expect(apiService.getScheduleUsage).toHaveBeenCalledWith(originalId)
      })

      test('Processingカバー表示中', () => {
        expect(
          storeModules.processing.mutations.endGet
        ).not.toHaveBeenCalled()
      })

      describe('予約団体マスタ取得：エラー', () => {
        beforeEach(() => {
          promiseMocker.setSwitch(loadReservationOrganizationsSwitch, false)
          promiseMocker.setSwitch(loadSwitch, true)
          apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
          promiseMocker.flushPendings()
        })

        test('エラー処理', () => {
          expect(apiService.isHandledError).toHaveBeenCalledWith(loadReservationOrganizationsError)
        })

        test('Processingカバー表示終了', () => {
          expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
        })
      })

      describe('コピー元データ取得処理：エラー', () => {
        beforeEach(() => {
          promiseMocker.setSwitch(loadReservationOrganizationsSwitch, true)
          promiseMocker.setSwitch(loadSwitch, false)
          apiService.isHandledError = jest.fn().mockReturnValueOnce(true)
          promiseMocker.flushPendings()
        })

        test('エラー処理', () => {
          expect(apiService.isHandledError).toHaveBeenCalledWith(loadError)
        })

        test('Processingカバー表示終了', () => {
          expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
        })
      })

      describe('データ取得：成功', () => {
        let makeFormArgs
        let form
        beforeEach(() => {
          promiseMocker.setSwitch(loadReservationOrganizationsSwitch, true)
          promiseMocker.setSwitch(loadSwitch, true)
          makeFormArgs = {
            name: originalRecord.name,
            is_public: originalRecord.is_public,
            reservation_organization_id: originalRecord.reservation_organization_id
          }
          form = myFormMocker.createMock(makeFormArgs)
          apiService.makeForm = jest.fn().mockReturnValueOnce(form)
          promiseMocker.flushPendings()
        })

        test('フォーム作成', () => {
          expect(apiService.makeForm).toHaveBeenCalledWith(makeFormArgs)
        })

        test('Processingカバー表示終了', () => {
          expect(storeModules.processing.mutations.endGet).toHaveBeenCalled()
        })

        describe('URL変更', () => {
          let newOriginalId
          beforeEach(() => {
            newOriginalId = originalId + 1
            setRouteCopyId(route, newOriginalId)
            apiService.getScheduleUsage = promiseMocker.mock(
              originalRecord,
              loadError,
              loadSwitch
            )
            apiService.getReservationOrganizationList = promiseMocker.mock(
              reservationOrganizations,
              loadReservationOrganizationsError,
              loadReservationOrganizationsSwitch
            )
            storeModules.processing.mutations.beginGet.mockReset()
            storeModules.processing.mutations.endGet.mockReset()
            router.push(route)
          })

          test('Processingカバー表示開始', () => {
            expect(storeModules.processing.mutations.beginGet).toHaveBeenCalled()
          })

          test('予約団体マスタ取得処理は呼び出さない', () => {
            expect(apiService.getReservationOrganizationList).not.toHaveBeenCalled()
          })

          test('コピー元データ取得処理を呼出', () => {
            expect(apiService.getScheduleUsage).toHaveBeenCalledWith(newOriginalId)
          })

          test('Processingカバー表示中', () => {
            expect(
              storeModules.processing.mutations.endGet
            ).not.toHaveBeenCalled()
          })
        })
      })
    })
  })
})
