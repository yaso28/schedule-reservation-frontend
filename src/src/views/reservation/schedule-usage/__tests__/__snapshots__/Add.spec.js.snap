// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add 通常 マウント metaタグ 1`] = `
NodeList [
  <meta
    content="noindex"
    data-vmid="robots"
    data-vue-meta="2"
    name="robots"
  />,
]
`;

exports[`Add 通常 マウント titleタグ 1`] = `"やそスポーツクラブ | 用途追加"`;

exports[`Add 通常 マウント データ取得：成功 ページ全体表示 1`] = `
<div>
  <h1>
    用途追加
  </h1>
   
  <form
    class=""
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="name"
          >
             名称 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <input
              class="form-control"
              id="name"
              type="text"
            />
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
         
        <div
          class="form-row form-group"
        >
          <label
            class="col-form-label col-sm-3"
            for=""
          >
            
    公開フラグ
  
          </label>
           
          <div
            class="col-sm-9"
          >
            <div
              class="my-input-wrapper"
            >
              <label
                class="c-switch form-check-label c-switch-primary c-switch-label"
              >
                <input
                  class="c-switch-input form-check-input"
                  id="is_public"
                  type="checkbox"
                />
                <span
                  class="c-switch-slider"
                  data-checked="✓"
                  data-unchecked="✕"
                />
              </label>
            </div>
             
            <!---->
          </div>
        </div>
         
        <div
          class="form-group form-row"
          role="group"
        >
          <label
            class="col-form-label col-sm-3"
            for="reservation_organization_id"
          >
             予約団体 
          </label>
          <div
            class="col-sm-9"
          >
            <!---->
            <select
              class="form-control"
              id="reservation_organization_id"
            >
              <!---->
              <option
                data-key="0"
                value=""
              >
                  
              </option>
              <option
                data-key="1"
                value="1"
              >
                 一般団体 
              </option>
              <option
                data-key="2"
                value="2"
              >
                 普及活動クラブ 
              </option>
            </select>
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
      </div>
       
      <footer
        class="card-footer"
      >
        <button
          class="btn btn-secondary"
          id="btn-reset"
          type="button"
        >
          
          リセット
        
        </button>
         
        <button
          class="btn btn-primary"
          type="submit"
        >
          
          保存
        
        </button>
      </footer>
    </div>
  </form>
</div>
`;
